pyuvdata/ms.py

```diff
@@ -0,0 +1,235 @@
+                self.telescope_location=np.array(np.mean(self.antenna_positions[np.invert(antFlags),:],axis=0))
+                #antenna names
+        ant_names=tbAnt.getcol('STATION')
+        self.Nants_telescope=len(antFlags[np.invert(antFlags)])
+        test_name=ant_names[0]
+        names_same=True
+        for antnum in range(len(ant_names)):
+            if(not(ant_names[antnum]==test_name)):
+                names_same=False
+        if(not(names_same)):
+            self.antenna_names=ant_names#cotter measurement sets store antenna names in the NAMES column. 
+        else:
+            self.antenna_names=tbAnt.getcol('NAME')#importuvfits measurement sets store antenna namesin the STATION column.
+        self.antenna_numbers=np.arange(len(self.antenna_names)).astype(int)
+        nAntOrig=len(self.antenna_names)
+        ant_names=[]
+        for antNum in range(len(self.antenna_names)):
+            if not(antFlags[antNum]):
+                ant_names.append(self.antenna_names[antNum])
+        self.antenna_names=ant_names
+        self.antenna_numbers=self.antenna_numbers[np.invert(antFlags)]
+        self.antenna_positions=self.antenna_positions[np.invert(antFlags),:]
+        '''
+        #remove blank names
+        for axnum in range(self.antenna_positions.shape[1]):
+            self.antenna_positions[:,axnum]-=np.mean(self.antenna_positions[:,axnum])
+        try:
+            thisTelescope=telescopes.get_telescope(self.instrument)
+            self.telescope_location_lat_lon_alt_degrees=(np.degrees(thisTelescope['latitude']),np.degrees(thisTelescope['longitude']),thisTelescope['altitude'])
+            #self.telescope_location=np.array(np.mean(tbAnt.getcol('POSITION'),axis=0))
+            print 'Telescope %s is known. Using stored values.'%(self.instrument)
+        except:
+            #If Telescope is unknown, use mean ITRF Positions of antennas
+            self.telescope_location=np.array(np.mean(tbAnt.getcol('POSITION'),axis=0))
+        '''
+        tbAnt.close()
+        tbField=tables.table(filepath+'/FIELD')
+        if(tbField.getcol('PHASE_DIR').shape[1]==2):
+            self.phase_type='drift'
+            self.set_drift()
+        elif(tbField.getcol('PHASE_DIR').shape[1]==1):
+            self.phase_type='phased'
+            self.phase_center_epoch=float(tb.getcolkeyword('UVW','MEASINFO')['Ref'][1:])#MSv2.0 appears to assume J2000. Not sure how to specifiy otherwise
+            self.phase_center_ra=float(tbField.getcol('PHASE_DIR')[0][0][0])
+            self.phase_center_dec=float(tbField.getcol('PHASE_DIR')[0][0][1])
+            self.set_phased()
+        #set LST array from times and itrf
+        self.set_lsts_from_time_array()
+
+        #set the history parameter
+        #as a string with \t indicating column breaks
+        #\n indicating row breaks.
+        self.history,self.casa_history=self._ms_hist_to_string(tables.table(filepath+'/HISTORY'),test_import_uvfits)
```

Comment:
Ok. I'm a bug fan of not losing information, especially if it provides provenance and/or context for the data. So let's keep it as a new optional keyword. We should take care to write it out to new formats if it exists so that if you read in a ms file and then write out a uvfits or miriad file and then read it back in you haven't lost the information.